lib/minidoc/finders.rb

```diff
@@ -24,17 +24,16 @@ def exists?(selector = {})
 
     def find(id_or_selector, options = {})
       if id_or_selector.is_a?(Hash)
-        options.merge!(transformer: method(:wrap))
-        collection.find(id_or_selector, options)
+        collection.find(id_or_selector, options).lazy.map(&method(:wrap))
```

Comment:
Consider code like this:

```rb
find(query).each do |repo|
  if repo.something?
    return repo
  end
end
```

WIth `transformer:`, the `wrap` method would only happen as each result is created and yielded. Without transformer (and without `.lazy`), you'd end up `to_a`ing all results, and `wrap`ing each result, even if you might only need a few before you `return` out of the loop.

I _think_ `.lazy` is going to work for us, provided the `Enumerator` contract is enough for a cursor to quack lazily as needed, without misbehaving if/when we did `find().count` or `find().sort()` The danger here is that a chained `count` or `sort` would become `Array#count` or `Array#sort` after we've `.lazy`d the cursor... but I'm not sure. Maybe there's a way to confirm with a spec?